### PR TITLE
feat(utils): Implement `get_size_hint` for default impls

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
Implements `get_size_hint` for the types that are implemented by default.

Currently, crates that call `u8.get_size_hint()` get the default value of 0 which makes it harder to implement `get_size_hint` for custom types, since they cannot simply call `get_size_hint` on the contained types recursively. This PR fixes that.

This change is non-breaking.

See also https://github.com/0xPolygonMiden/miden-base/pull/895#pullrequestreview-2326556758 for more context.

### CODE REVIEW

File: utils/core/src/serde/byte_writer.rs [hunk 1]
```diff
@@ -141,9 +141,9 @@ impl ByteWriter for alloc::vec::Vec<u8> {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-/// Returns the length of the value in vint64 encoding.
-fn encoded_len(value: u64) -> usize {
-    let zeros = value.leading_zeros() as usize;
+/// Returns the length of the usize value in vint64 encoding.
+pub(super) fn usize_encoded_len(value: usize) -> usize {
```
Comment (irakliyk) [inline utils/core/src/serde/byte_writer.rs:145]: Passing value as `usize` into here would cause issues on 32-bit systems (e.g., WASM). Let's revert this change.

**Review by irakliyk — COMMENTED** [review]
Thank you! I left one comment inline - but other than that, it looks great.

File: utils/core/src/serde/byte_writer.rs (hunk 1 above)
Comment (PhilippGackstatter) [reply utils/core/src/serde/byte_writer.rs]: I see, good point!
Updated to cast the usize in the `get_size_hint` function to a u64 instead.

**Review by irakliyk — APPROVED** [review]
All looks good! Thank you!
